Matrix.py

```diff
@@ -0,0 +1,168 @@
+"""matrices calculator class."""
+
+
+class Matrix:
+    """A matrices calculator for basic math actions
+        Attributes:
+            data(tuple of tuples): a Matrix- each line is a tuple in the major tuple """
+
+    def __init__(self, data):
+        if type(data) is not tuple:
+            raise ValueError("Invalid matrix")
+        for line in data:
+            if len(line) != len(data) or type(line) is not tuple:
+                raise ValueError("Invalid matrix")
+        self.data = data
+
+    def __iter__(self):
+        return iter(self.data)
+
+    def __getitem__(self, item):
+        return self.data[item]
+
+    def __repr__(self):
+        """prints the matrix in a format that can be used for build another Matrix"""
+        return f"{self.__class__.__name__}({self.data})"
+
+    def __str__(self):
+        matrix_in_lines = ""
+        for i in range(len(self.data)):
+            matrix_in_lines += (str(self.data[i]) + "\n")
+        return matrix_in_lines
+
+    @staticmethod
+    def unity(size):
+        """ unity method creates a matrix of zeros with a diagonal of ones
+        args: size(int)- the size of the matrix and lines.
+        returns: a matrix full of 0 and 1's diagonal
+        exceptions: ValueError for non-ints"""
+        if type(size) is int:
+            major_list = [[1 if i == index else 0 for i in range(size)] for index in range(size)]
+            # converts list of lists to type Matrix. lt are the inside lists
+            return Matrix((tuple([tuple(lt) for lt in major_list])))
+        else:
+            raise ValueError("size must be an integer")
+
+    @staticmethod
+    def ones(size):
+        """ones method returns a Matrix full of ones whose size determined by size variable
+        args: size(int)- the size of the matrix and the lines
+        returns: a matrix full of ones
+        exceptions: ValueError for non-ints"""
+        if type(size) is int:
+            sub_list = [tuple(lt) for lt in [([1] * size) for i in range(size)]]
+            matrix_to_return = Matrix((tuple(sub_list)))
+            return matrix_to_return
+        else:
+            raise ValueError("size must be an integer")
+
+    def __add__(self, other):
+        """add a matrix to another one
+            args: self.data (tuple)- A matrix
+                other(tuple) -A matrix
+
+            returns: return the new matrix (tuple)
+
+             raises: syntaxerror if "other" isn't a matrix"""
+        if type(other) is Matrix:
+            answer = []
+            for line in range(len(self.data)):
```

Comment:
why not iterate over an item in `self.data`?
```suggestion
            for line in self.data:
```
I don't think you need to iterate over indexes